keccak256/src/circuit.rs

```diff
@@ -0,0 +1,284 @@
+            theta_config,
+            rho_config,
+            pi_config,
+            xi_config,
+            iota_b9_config,
+            iota_b13_config,
+            absorb_config,
+            state,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn assign_state(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        state: [F; 25],
+    ) -> Result<[F; 25], Error> {
+        // In case is needed
+        let mut state = state;
+        let mut offset = offset;
+
+        // First 23 rounds
+        for round in 0..24 {
+            // State in base-13
+            // theta
+            state = {
+                // Apply theta outside circuit
+                let out_state = KeccakFArith::theta(&state_to_biguint(state));
+                let out_state = state_bigint_to_pallas(out_state);
+                // assignment
+                self.theta_config
+                    .assign_state(region, offset, state, out_state)?
+            };
+
+            offset += 1;
+
+            // rho
+            state = {
+                // Apply rho outside circuit
+                let out_state = KeccakFArith::rho(&state_to_biguint(state));
+                let out_state = state_bigint_to_pallas(out_state);
+                // assignment
+                self.rho_config
+                    .assign_state(region, offset, state, out_state)?
+            };
+            // Outputs in base-9 which is what Pi requires.
+            offset += 1;
+
+            // pi
+            state = {
+                // Apply pi outside circuit
+                let out_state = KeccakFArith::pi(&state_to_biguint(state));
+                let out_state = state_bigint_to_pallas(out_state);
+                // assignment
+                self.pi_config
+                    .assign_state(region, offset, state, out_state)?
+            };
+
+            offset += 1;
+
+            // xi
+            state = {
+                // Apply xi outside circuit
+                let out_state = KeccakFArith::xi(&state_to_biguint(state));
+                let out_state = state_bigint_to_pallas(out_state);
+                // assignment
+                self.xi_config
+                    .assign_state(region, offset, state, out_state)?
+            };
+
+            offset += 1;
+
+            // iota_b9
+            state = {
+                // Apply iota_b9 outside circuit
+                let out_state = KeccakFArith::iota_b9(
+                    &state_to_biguint(state),
+                    ROUND_CONSTANTS[round],
+                );
+                let out_state = state_bigint_to_pallas(out_state);
+                // assignment
+                self.iota_b9_config
+                    .assign_state(region, offset, state, out_state)?
+            };
+            // The resulting state is in Base-13 now. Which is what Theta requires again at the start of the loop.
+            offset += 1;
+        }
+        let round = 24;
```

Comment:
Same here. We can get the constant 24 from PERMUTATION in common.rs

Reply:
ditto!

Thanks!!!